ffc/process_charge_file.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import logging
+import tempfile
+from typing import Any
+
+from django.conf import settings
+
+from ffc.client import get_httpx_ffc_api_client
+
+logger = logging.getLogger(__name__)
+
+
+class ChargeFileSplitProcessor:
+    def __init__(
+        self,
+        mtp_base_api_url: str,
+        max_concurrent_requests: int = settings.EXTENSION_CONFIG.get(
+            "FFC_BILLING_PROCESS_MAX_CONCURRENCY", "10"
+        ),
+    ):
+        self.finops_client = get_httpx_ffc_api_client()
+        self.MPT_API_BASE_URL = mtp_base_api_url
+        self.semaphore = asyncio.Semaphore(int(max_concurrent_requests))
+
+    async def process_charge_file(
+        self, charge_file_id: str, download_folder: str
+    ) -> str:
+        async with self.semaphore:
+            # first download the zip file related to the given charge_file_id
+            zip_file_path = await self.finops_client.get_charges_file_download_url(
+                charge_file_id=charge_file_id, download_folder=download_folder
+            )
+            if zip_file_path is None:
+                logger.error(
+                    f"No file was downloaded for the charge file: {charge_file_id} ."
+                )
+            return zip_file_path  # todo: this will be the input for the next operation
```

Comment:
The next operation should be invoked from this method, so there is no need to return it